### PR TITLE
[sw/testing] organize chip-level test code - part 3/6

### PR DESCRIPTION
**Pull request by timothytrippel**
This addresses the third item in #7976 under "Summary of Desired Changes to sw/device/tests".

This PR depends on #8000 and #8039, thus has duplicate commits that will be removed when #8000 and #8039 are merged.

### CODE REVIEW

**Comment by timothytrippel** [comment]
One PR dependency (#8000) merged and duplicate commits removed. One more PR (#8039) dependency still remains.

**Comment by timothytrippel** [comment]
Final PR dependency (#8039) merged. Ready for review.

**Review by moidx — APPROVED** [review]

**Review by cindychip — APPROVED** [review]

File: hw/top_earlgrey/dv/chip_dif_tests.hjson [hunk 1]
```diff
@@ -56,18 +56,6 @@
       sw_images: ["sw/device/tests/kmac_smoketest:1"]
       en_run_modes: ["sw_test_mode"]
     }
-    {
```
Comment (sriyerg) [inline hw/top_earlgrey/dv/chip_dif_tests.hjson:59]: Why are these removed instead of being renamed?

File: hw/top_earlgrey/dv/chip_dif_tests.hjson [hunk 2]
```diff
@@ -56,18 +56,6 @@
       sw_images: ["sw/device/tests/kmac_smoketest:1"]
```
Comment (sriyerg) [inline hw/top_earlgrey/dv/chip_dif_tests.hjson:56]: This file needs to be renamed to `chip_dif_smoketests.hjson`.

File: hw/top_earlgrey/dv/chip_dif_tests.hjson (hunk 1 above)
Comment (timothytrippel) [reply hw/top_earlgrey/dv/chip_dif_tests.hjson:59]: PTAL at #7976, specifically the portion that says "There will be one KMAC smoke test..."  and the "Organizing Chip-Level Test Code" design doc that was discussed and approved at the Si WG two weeks ago.

File: hw/top_earlgrey/dv/chip_dif_tests.hjson (hunk 2 above)
Comment (timothytrippel) [reply hw/top_earlgrey/dv/chip_dif_tests.hjson:56]: Added this a task under "Summary of Desired Changes to sw/device/tests" in #7976, since there are still tasks to be completed to close out #7976. Also, I suggest renaming to `chip_smoketests.hjson` rather than `chip_dif_smoketests.hjson`, since all chip-level tests use DIFs, and to stay consistent with the other changes proposed in the "Organizing Chip-Level Test Code" design doc.

File: hw/top_earlgrey/dv/chip_dif_tests.hjson (hunk 1 above)
Comment (sriyerg) [reply hw/top_earlgrey/dv/chip_dif_tests.hjson:59]: Right, but the tests exist - they are just renamed. They need to be added back to the adjoining `chip_sim_cfg.hjson`.

Comment (timothytrippel) [reply hw/top_earlgrey/dv/chip_dif_tests.hjson:59]: Added this a task under "Summary of Desired Changes to sw/device/tests" in #7976, and will complete in a future PR to close out #7976.
